client/service/src/chain_ops.rs

```diff
@@ -116,55 +288,32 @@ impl<
 			let client = &self.client;
 			let queue = &mut self.import_queue;
 
-			// Start by reading the number of blocks if not done so already.
```

Comment:
Since you're already rewriting this block, I think this could have been rewrited as an `async` block instead, as it would make things way more clear. I'm not sure how familiar you are with `async` though.

Reply:
It was part of the initial plan, and then dropped because I wanted to get this in last week, but now that there is no time-constraints anymore I think it's a good idea. I'm not familiar with `async` blocks, but I won't learn if I never try! :)

Reply:
Now that I think about it, wouldn't it make more sense to have this on a different PR?

Reply:
Yes. I think I tried once and it was not trivial. So it might be better to do that in another PR so you don't accumulate more merge conflicts (imo)